pipeline/lib/json_stream.ml

```diff
@@ -38,27 +42,38 @@ let json_step stack chr =
   | _, (('{' | '[' | '"') as chr) -> chr :: stack
   | _ -> stack
 
-let make_json_parser () = { current = Buffer.create 16; stack = [] }
+let make_json_parser () =
+  { current = Buffer.create 16; stack = [ '\n' ]; lines = 1; start_line = 1 }
 
 let json_step state chr =
+  let state =
+    match chr with '\n' -> { state with lines = state.lines + 1 } | _ -> state
+  in
   match json_step state.stack chr with
   | [] ->
-      if Buffer.length state.current = 0
-      then (None, { state with stack = [] })
+      if Buffer.length state.current = 0 then (None, { state with stack = [] })
       else (
         Buffer.add_char state.current chr;
         let str = Buffer.contents state.current in
-        (Some str, make_json_parser ()))
+        let st = make_json_parser () in
+        ( Some str,
+          { st with lines = state.lines; start_line = state.start_line } ))
   | hd :: _ as stack ->
-      if hd <> '\n' then Buffer.add_char state.current chr;
-      (None, { state with stack })
+      if chr <> '\n' || hd = '"' then Buffer.add_char state.current chr;
+      let start_line =
+        if Buffer.length state.current == 1 then state.lines
```

Comment:
It took me a bit of time to understand the `1` and not `0`, but that's because we `add_char` on the line above... Perhaps `if state.stack = []` is a simpler criteria independent of side effects? (meaning that the parser was not parsing before, but now is)

(small typo: it's a single `=` in caml, even if `==` works fine for integers)

Reply:
Yes, checking the `stack` state before and now seems nicer, indeed! thanks